# Competition configuration options

options:
  insert_models_over_bins: true
  fill_demo_tray: false

belt_parts:
  piston_rod_part:
    3.0:
      pose:
        xyz: [0.0, 0.0, 0.1]
        rpy: [0, 0, 'pi/2']
  part4:
    6.0:
      pose:
        xyz: [0.0, 0.0, 0.1]
        rpy: [0, 0, 'pi/2']

orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    parts:
      part_0:
        type: piston_rod_part
        pose:
          xyz: [0.1, 0, 0]
          rpy: [0, 0, 0]
      part_1:
        type: piston_rod_part
        pose:
          xyz: [-0.1, 0, 0]
          rpy: [0, 0, 'pi']

models_over_bins:
  bin7:
    models:
      piston_rod_part:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 3
        num_models_y: 4

faulty_parts:
  - piston_rod_part_5  # this part gets spawned manually in the tutorial

time_limit: -1


# Competitor trial configuration options

arm:
  type: ur10

sensors:
  break_beam:
    type: break_beam
    pose:
      xyz: [1.6, 2.25, 0.91]
      rpy: [0, 0, 'pi']
  proximity_sensor:
    type: proximity_sensor
    pose:
      xyz: [0.95, 2.6, 0.916]
      rpy: [0, 0, 0]
  logical_camera:
    type: logical_camera
    pose:
      xyz: [-0.66, 0.22, 1.36]
      rpy: ['pi/2', 1.32, 0]
  logical_camera_over_agv1:
    type: logical_camera
    pose:
      xyz: [0.3, 3.15, 1.5]
      rpy: [0, 'pi/2', 0]
  laser_profiler:
    type: laser_profiler
    pose:
      xyz: [1.21, 4, 1.64]
      rpy: ['-pi', 'pi/2', 'pi/2']
